experimental/dds/tree2/src/core/undo/undoRedoManager.ts

```diff
@@ -196,11 +252,14 @@ export interface UndoableCommit<TChange> {
 	readonly parent?: UndoableCommit<TChange>;
 }
 
+interface RedoableCommit<TChange> extends UndoableCommit<TChange> {}
+
 /**
  * The type of a commit in the context of undo/redo manager.
  */
 export enum UndoRedoManagerCommitType {
 	Undoable,
+	Redoable,
```

Comment:
I'd say this enum is now a candidate for a [flags enum](https://www.typescriptlang.org/play?#code/KYOwrgtgBAqiAmB7ASsJBhREIEsAuAKgJ4AOwUA3gFBS2wKICGARgDbkC8UAjFADx8oABgA0NOqiQt2ULrwE8xdeklk9+ggExQAPlElM2wJRLSI18wQGZdKw+zEBfKlQDGiEAGc8UMA2mcdgaY2PjEZAB0cFJGANxuHt5QAE5mAWrRKGYhuISkwBEGAfHuXj5+qlyZwVi54QWZJYk+qZVB2bVh+YVm8VQAZn6ueDgeUDiemQEAFHj5AFztGJ15ZACUi8yIiOyMIJTitKl4YMn7s-lQAGRLiDldkVNGa1AAhBxcQvHOA0MjYxMikYLmRFtUOqFVsANlAtjtgHsDspjqdznMyNdbvcoT0YuwXu9Pt8XKVPPCIqxEABzaYTJ7saYVezQtZNLzkyk0150-zA1rMtashLs9gU6nTbmTXkMpmCtlk0Wc2lSvHAab8uXChUFJWSoEy6Us+Uc8WAtJ8834oWkk00s2qxkMTU2xXivWWtUa1lAA), since the states overlap (Redo is also Undoable, Undo is also Redoable). The benefit of doing so is that the "overlappingness" is encoded into the enum values themselves, so you only ever need to check a value against one enum value (via the bit check), rather than checking against multiple enum values (e.g. "is this undoable? becomes a single bit check rather than iterating through all enum values that are "undoable" and checking each one). It doesn't buy us a whole lot here because this enum is still small and you only ever have to check two values at once, but you can see how this would help your logic scale if you added more types to your enum. I just thought I'd point it out since I think it's a cool pattern and kind of fun to write.

Reply:
I like the idea but I'm gonna pass on it (at least for now) cause it doesn't improve anything currently and it makes the linter angry 😄.

Reply:
I'm not surprised, linters ruin all the fun :(